pkg/runtime/podman.go

```diff
@@ -250,32 +251,28 @@ func (p *PodmanRuntime) ReadLogBuffer(unit string, n int) ([]logbuf.LogEntry, er
 	yes := true
 	tail := strconv.Itoa(n)
 	out := make(chan string)
-	wg := &sync.WaitGroup{}
-	wg.Add(1)
 	opts := containers.LogOptions{
 		Stderr:     &yes,
 		Stdout:     &yes,
 		Tail:       &tail,
 		Timestamps: &yes,
 	}
 	var logs []logbuf.LogEntry
-	go func(wg *sync.WaitGroup) {
+	go func() {
 		err := containers.Logs(p.connText, containerName, opts, out, out)
 		if err != nil {
 			glog.Errorf("cannot get logs for container %s : %v", containerName, err)
 		}
-		wg.Done()
 		close(out)
-	}(wg)
+	}()
 	for msg := range out {
 		logLine := strings.Split(msg, " ")
 		logs = append(logs, logbuf.LogEntry{
 			Timestamp: logLine[0],
 			Source:    logbuf.StdoutLogSource,
-			Line:      strings.Join(logLine[1:], " "),
+			Line:      strings.Join(logLine[1:], "") + "\n",
```

Comment:
Maybe check that `logLine` has at least two elements?

Reply:
:+1: fixed